### ISSUE TITLE
good suggestion for debugging jemalloc and output result to somewhere

### ISSUE THREAD

**Issue by wenqicao** [jemalloc/jemalloc#1283]
I am now modifying jemalloc to do some test. I wanna output some variable result to stdout. Simply testing `printf` doesn't work. any suggestions to do that?

**Comment by davidtgoldblatt** [comment]
There's malloc_printf, which exposes a reasonable chunk of the printf API. You might also want the logging interface, which can let individual statements be turned on or off depending on malloc_conf settings.

**Comment by wenqicao** [comment]
`malloc_printf` works pretty good! awesome! 

BTW, one more quick question, I wanna create another heap to track fullruns (I am not sure if you still remember we've discussed that). 
I used `arena_run_heap_new( )` to init the heap and use `arena_run_heap_insert( )` `arena_run_heap_remove( )` to as insert/remove operation, but met segmentation fault. Do you have idea about where I was wrong? or which part I missed, or which part I need to pay attention?
Thanks

**Comment by davidtgoldblatt** [comment]
Yeah, I believe that that's all you should need to do. (And of course, ensure that you hold the appropriate locks during modifications, make sure you remove from one heap before inserting in another, etc.).

**Comment by stan1199** [comment]
@davidtgoldblatt can you please point me to a doc or something which mentions how to enable internal debug logs of jemalloc? I couldn't find it. Also, I assume it logs to stderr right?
